### ISSUE TITLE
Funimation Extractor not working with age restricted shows

### ISSUE THREAD

**Issue by nmills3** [ytdl-org/youtube-dl#29256]
## Checklist

- [ x] I'm reporting a broken site support
- [x ] I've verified that I'm running youtube-dl version **2021.06.06**
- [x ] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x ] I've searched the bugtracker for similar issues including closed ones

## Verbose log

C:\Users\nicho\AppData\Roaming\nm-yt-downloader>youtube-dl.exe --cookies funimation.com_cookies.txt https://www.funimation.com/en/shows/mushoku-tensei-jobless-reincarnation/jobless-reincarnation/ -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--cookies', 'funimation.com_cookies.txt', 'https://www.funimation.com/en/shows/mushoku-tensei-jobless-reincarnation/jobless-reincarnation/', '-v']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3
[debug] Proxy map: {}
[Funimation] jobless-reincarnation: Downloading webpage
[Funimation] 1376136: Downloading JSON metadata
ERROR: An extractor error has occurred. (caused by KeyError('items',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type youtube-dl -U to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\extractor\common.py", line 534, in extract
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\extractor\funimation.py", line 121, in _real_extract
KeyError: 'items'
Traceback (most recent call last):
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\extractor\common.py", line 534, in extract
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\extractor\funimation.py", line 121, in _real_extract
KeyError: 'items'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 815, in wrapper
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\YoutubeDL.py", line 836, in __extract_info
File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpkqxnwl31\build\youtube_dl\extractor\common.py", line 547, in extract
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('items',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type youtube-dl -U to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

## Description

The funimation extractor is not working on shows that require an account with mature content restriction turned off even when providing the cookies for a valid account with mature content restriction turned off

**Comment by joshbrown50** [comment]
Why was this closed? I'm having the same issue

**Comment by joshbrown50** [comment]
And were you able to find a fix? It seems like it has to do with episodes that require a login to download. I can find a bunch of related issues/cases, but I haven't seen anyone suggest a fix for this.

**Comment by Jules-A** [comment]
It's a duplicate of a bug from 2019 or something... I was getting this but making sure the cookies aren't older than a few hours and making sure I close the Funimation website before starting the download seemed to fix it for me. I also provide login info but I'm not sure that gets used if I supply cookies.

**Comment by joshbrown50** [comment]
That's not working for me. I log in to funimation. Get the cookies file. close the tab. and try it and it fails. I wait a bit and it still fails.
I keep getting an error that says No valid Free Avails for this experience

**Comment by Jules-A** [comment]
> 
> 
> That's not working for me. I log in to funimation. Get the cookies file. close the tab. and try it and it fails. I wait a bit and it still fails.
> I keep getting an error that says No valid Free Avails for this experience

Are you also providing log in details? I'm using Netrc for them as well as cookies but I also add my browser user-agent in the command. It could also be the method you are using to extract cookies isn't working or the browser is keeping the session alive even after closing the tab? I use Firefox so not sure about other browsers and I also only export the cookies for the funimation domain instead of all cookies.

**Comment by joshbrown50** [comment]
I'm also using netrc and cookies and my browser's user agent, and I also closed firefox entirely before I ran it and I'm still getting this error

**Comment by Jules-A** [comment]
Hmm... You could show your verbose log I guess but I'm not so sure the Youtube-dl devs really care. I'm actually using [yt-dlp](https://github.com/yt-dlp/yt-dlp) but as far as I'm aware the Funimation extractor is the same.
The commandline I'm using is specific to yt-dlp but this is what the verbose looks like for me: 

```[debug] Command-line config: ['-v', '-a', 'ToDownload - akito.txt', '-o', '%(extractor)s\\%(title)s - %(myindex)s.%(ext)s', '--parse-metadata', 'original_url:#%(myindex)s', '--parse-metadata', '%(series)s - S%(season_number)sE%(episode_number)s - %(episode)s:^(?P<title>.+ - S\\d+E\\d+ - .+)$', '--cookies', 'cookies-funimation-com.txt', '--output-na-placeholder', '', '--embed-subs', '--sub-langs', 'en,enUS', '--user-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:89.0) Gecko/20100101 Firefox/89.0', '-n', '--fragment-retries', 'infinite', '-N', '3', '--no-mtime', '--download-archive', 'archive.txt', '-P', 'temp:D:\\Temp\\']
[debug] Batch file urls: ['https://www.funimation.com/en/shows/code-geass-akito-the-exiled/episode-1-the-wyvern-arrives#1', 'https://www.funimation.com/en/shows/code-geass-akito-the-exiled/episode-2-the-wyvern-divided#2', 'https://www.funimation.com/en/shows/code-geass-akito-the-exiled/episode-3-the-brightness-falls#3', 'https://www.funimation.com/en/shows/code-geass-akito-the-exiled/episode-4-memories-of-hatred#4', 'https://www.funimation.com/en/shows/code-geass-akito-the-exiled/final-episode-to-beloved-ones#5']
[debug] Loading archive file 'archive.txt'

[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.06.01 (exe)
[debug] Python version 3.9.5 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg N-102688-gcf1746d779, ffprobe N-102688-gcf1746d779
[debug] Proxy map: {}
[Funimation] Logging in
[debug] [Funimation] Extracting URL: https://www.funimation.com/en/shows/code-geass-akito-the-exiled/episode-1-the-wyvern-arrives#1
[Funimation] episode-1-the-wyvern-arrives: Downloading webpage
[Funimation] 352430: Downloading JSON metadata
[Funimation] 352430: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Searching for r"\#(?P<myindex>.+)" in %(original_url)s
[MetadataFromField] parsed myindex from "original_url": 1
[debug] Searching for r"^(?P<title>.+ - S\d+E\d+ - .+)$" in %(series)s - S%(season_number)sE%(episode_number)s - %(episode)s
[MetadataFromField] parsed title from "%(series)s - S%(season_number)sE%(episode_number)s - %(episode)s": Code Geass - Akito the Exiled - S1E1 - EPISODE 1: The Wyvern Arrives
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 352430: Downloading 1 format(s): hls-10013
[debug] Invoking downloader on "https://vmfst-api.prd.funimationsvc.com/FunimationStoreFront/1621110/English/1621110_English_82c849c2-a765-e711-8175-020165574d09_Layer10.m3u8?Key-Pair-Id=APKAIHNXECY27H4O6NIA&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9kMzNldDc3ZXZkOWJnZy5jbG91ZGZyb250Lm5ldC9GdW5pbWF0aW9uU3RvcmVGcm9udC8xNjIxMTEwL0VuZ2xpc2gvKiIsIkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTYyNDExNjU2OH19fV19&Signature=Id71lFMQdOTyvuDaHO4Obnimj9B5GJMgwqQqslXXUBmSZkx14Zsln97Jem9UxA07HyRbKJrGhq6ee-ahT4iiL8E2MNkVkOkneXnqdbf4K9C1yLd3D7E3oPKuotU5EmnJnoxrkiBnwIponMJ4N6~VHllKGqEiAKT8xAbAKNCHLWUv0o-u81R2xRio3~3LaTAnrCZ83oTQtt5LBEpTaV6WsiHhMfJ8FhMkCy49HWIot7USNJtZ~-kJRefYTNY0b4GYChT80ub7kkGTwye4~so4QsAnbCZ~imbHZzFFHtav22pBQ2PliCStwQdoOp3yqH8bb~iWhGiz1FGxLLTMLC~mmQ__&rt=3069394"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 384
[download] Destination: D:\Temp\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.mp4
WARNING: The download speed shown is only of one thread. This is a known issue
[download] 100% of 2.83GiB in 12:48
[debug] ffmpeg command line: ffprobe -show_streams "file:D:\Temp\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.mp4"
[FixupM3u8] Fixing malformed AAC bitstream in "D:\Temp\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:D:\Temp\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.mp4" -c copy -map 0 -dn -f mp4 "-bsf:a" aac_adtstoasc "file:D:\Temp\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.temp.mp4"
[EmbedSubtitle] There aren't any subtitles to embed
[MoveFiles] Moving file "D:\Temp\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.mp4" to "D:\Programs\YoutubeDL\Funimation\Code Geass - Akito the Exiled - S1E1 - EPISODE 1 - The Wyvern Arrives - 1.mp4"```

**Comment by joshbrown50** [comment]
C:\Windows\system32>youtube-dl https://www.funimation.com/en/shows/the-saints-magic-power-is-omnipotent/predicament/
[debug] System config: []
[debug] User config: ['--cookies', 'D:UsersUserDownloadsyoutube_dlcookies.txt', '--user-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:88.0) Gecko/20100101 Firefox/88.0', '-o', 'D:\\Users\\User\\Downloads\\youtube_dl\\Downloads\\%(series)s-%(title)s.%(ext)s', '--netrc', '--verbose']
[debug] Custom config: []
[debug] Command-line args: ['https://www.funimation.com/en/shows/the-saints-magic-power-is-omnipotent/predicament/']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.7.0 (CPython) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[Funimation] Logging in
[Funimation] predicament: Downloading webpage
[Funimation] 1609301: Downloading JSON metadata
Sources: {
    "errors": [
        {
            "code": 4,
            "detail": "No valid Free Avails for this experience",
            "title": "No avails"
        }
    ]
}
ERROR: An extractor error has occurred. (caused by KeyError('items')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\extractor\common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\extractor\funimation.py", line 124, in _real_extract
    sources = sources['items']
KeyError: 'items'
Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\extractor\common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\extractor\funimation.py", line 124, in _real_extract
    sources = sources['items']
KeyError: 'items'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37-32\lib\site-packages\youtube_dl-2021.6.6-py3.7.egg\youtube_dl\extractor\common.py", line 547, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('items')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

**Comment by Jules-A** [comment]
From what I can see the cookies file is missing backslashes so not being used.

**Comment by joshbrown50** [comment]
So weirdly enough, there were backslashes in my config file, but I decided to put 2 slashes for every one and now its working

**Comment by joshbrown50** [comment]
Thanks a ton for helping me!
